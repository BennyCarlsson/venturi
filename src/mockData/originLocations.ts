export const locations = {
  locations: {
    loadingOrigin: false,
    loadingDestination: false,
    destinationLocations: [
      {
        id: '9021014014010000',
        idx: '1',
        name: 'Hålta kyrka, Kungälv',
        lat: '57.895206',
        lon: '11.827647',
        type: 0,
      },
      {
        id: '9021014012743000',
        idx: '2',
        name: 'Gårda, Mölndal',
        lat: '57.565347',
        lon: '12.092784',
        type: 0,
      },
      {
        id: '9021014002515000',
        idx: '3',
        name: 'Toftaåsgatan, Göteborg',
        lat: '57.645953',
        lon: '11.917170',
        type: 0,
      },
      {
        id: '9021014015934000',
        idx: '4',
        name: 'Hagaån, Orust',
        lat: '58.139155',
        lon: '11.625866',
        type: 0,
      },
      {
        id: '9021014021670000',
        idx: '5',
        name: 'Råane bro, Uddevalla',
        lat: '58.293814',
        lon: '11.984122',
        type: 0,
      },
      {
        id: '9021014021673000',
        idx: '6',
        name: 'Råane skola, Uddevalla',
        lat: '58.288996',
        lon: '11.988985',
        type: 0,
      },
      {
        id: '9021014021672000',
        idx: '7',
        name: 'Råane södra, Uddevalla',
        lat: '58.291360',
        lon: '11.986549',
        type: 0,
      },
      {
        id: 'A U Bååths väg, 439 94 Onsala',
        idx: '8',
        name: 'A U Bååths väg, 439 94 Onsala',
        lat: '57.390569',
        lon: '12.022587',
        type: 1,
      },
      {
        id: 'Allstakan Flaåsen, 671 98 Gunnarskog',
        idx: '9',
        name: 'Allstakan Flaåsen, 671 98 Gunnarskog',
        lat: '59.908834',
        lon: '12.510476',
        type: 1,
      },
      {
        id: 'Almvägen, 432 79 Tvååker',
        idx: '10',
        name: 'Almvägen, 432 79 Tvååker',
        lat: '57.044575',
        lon: '12.408116',
        type: 1,
      },
      {
        id: 'Alvägen, 432 79 Tvååker',
        idx: '11',
        name: 'Alvägen, 432 79 Tvååker',
        lat: '57.042903',
        lon: '12.404539',
        type: 1,
      },
      {
        id: 'Amandas väg, 432 77 Tvååker',
        idx: '12',
        name: 'Amandas väg, 432 77 Tvååker',
        lat: '57.060683',
        lon: '12.529336',
        type: 1,
      },
      {
        id: 'Anders Kristians väg, 432 77 Tvååker',
        idx: '13',
        name: 'Anders Kristians väg, 432 77 Tvååker',
        lat: '57.073115',
        lon: '12.368690',
        type: 1,
      },
      {
        id: 'Anemonvägen, 432 76 Tvååker',
        idx: '14',
        name: 'Anemonvägen, 432 76 Tvååker',
        lat: '57.012168',
        lon: '12.352572',
        type: 1,
      },
      {
        id: 'Angryd, 432 77 Tvååker',
        idx: '15',
        name: 'Angryd, 432 77 Tvååker',
        lat: '57.089359',
        lon: '12.540393',
        type: 1,
      },
      {
        id: 'Annelundsvägen, 432 76 Tvååker',
        idx: '16',
        name: 'Annelundsvägen, 432 76 Tvååker',
        lat: '57.042570',
        lon: '12.315941',
        type: 1,
      },
      {
        id: 'Askvägen, 432 79 Tvååker',
        idx: '17',
        name: 'Askvägen, 432 79 Tvååker',
        lat: '57.043370',
        lon: '12.403046',
        type: 1,
      },
      {
        id: 'Aspvägen, 432 79 Tvååker',
        idx: '18',
        name: 'Aspvägen, 432 79 Tvååker',
        lat: '57.044215',
        lon: '12.406894',
        type: 1,
      },
      {
        id: 'Augusts väg, 432 77 Tvååker',
        idx: '19',
        name: 'Augusts väg, 432 77 Tvååker',
        lat: '57.058598',
        lon: '12.526792',
        type: 1,
      },
      {
        id: 'Avavägen, 432 78 Tvååker',
        idx: '20',
        name: 'Avavägen, 432 78 Tvååker',
        lat: '57.047721',
        lon: '12.397797',
        type: 1,
      },
      {
        id: 'Rör, 432 77 Tvååker',
        idx: '21',
        name: 'Rör, 432 77 Tvååker',
        lat: '57.059937',
        lon: '12.445323',
        type: 1,
      },
      {
        id: 'Sik, 432 76 Tvååker',
        idx: '22',
        name: 'Sik, 432 76 Tvååker',
        lat: '57.001220',
        lon: '12.367845',
        type: 1,
      },
      {
        id: 'Påarps gård, Håcksvik',
        idx: '23',
        name: 'Påarps gård, Håcksvik',
        lat: '57.315473',
        lon: '13.147488',
        type: 1,
      },
      {
        id: 'Ängarna, 432 76 Tvååker',
        idx: '24',
        name: 'Ängarna, 432 76 Tvååker',
        lat: '57.028376',
        lon: '12.375692',
        type: 1,
      },
      {
        id: 'Ängvägen, 432 78 Tvååker',
        idx: '25',
        name: 'Ängvägen, 432 78 Tvååker',
        lat: '57.047217',
        lon: '12.402363',
        type: 1,
      },
      {
        id: 'Ästad, 432 77 Tvååker',
        idx: '26',
        name: 'Ästad, 432 77 Tvååker',
        lat: '57.066535',
        lon: '12.562155',
        type: 1,
      },
      {
        id: 'Ästad bondgård, 432 77 Tvååker',
        idx: '27',
        name: 'Ästad bondgård, 432 77 Tvååker',
        lat: '57.063245',
        lon: '12.555953',
        type: 1,
      },
      {
        id: 'Bäckvägen, 432 79 Tvååker',
        idx: '28',
        name: 'Bäckvägen, 432 79 Tvååker',
        lat: '57.038741',
        lon: '12.404889',
        type: 1,
      },
      {
        id: 'Ballstorp Kullaås, 535 93 Kvänum',
        idx: '29',
        name: 'Ballstorp Kullaås, 535 93 Kvänum',
        lat: '58.191256',
        lon: '13.253812',
        type: 1,
      },
      {
        id: 'Barrvägen, 432 79 Tvååker',
        idx: '30',
        name: 'Barrvägen, 432 79 Tvååker',
        lat: '57.035675',
        lon: '12.405806',
        type: 1,
      },
      {
        id: 'Beckasinvägen, 432 76 Tvååker',
        idx: '31',
        name: 'Beckasinvägen, 432 76 Tvååker',
        lat: '57.007296',
        lon: '12.358487',
        type: 1,
      },
      {
        id: 'Bergsjö Lövbodaåsarna, 681 94 Kristinehamn',
        idx: '32',
        name: 'Bergsjö Lövbodaåsarna, 681 94 Kristinehamn',
        lat: '59.421358',
        lon: '14.248622',
        type: 1,
      },
      {
        id: 'Bjällemad, 432 79 Tvååker',
        idx: '33',
        name: 'Bjällemad, 432 79 Tvååker',
        lat: '57.037329',
        lon: '12.411271',
        type: 1,
      },
      {
        id: 'Bjällemadvägen, 432 79 Tvååker',
        idx: '34',
        name: 'Bjällemadvägen, 432 79 Tvååker',
        lat: '57.037662',
        lon: '12.406678',
        type: 1,
      },
      {
        id: 'Bjären, 432 76 Tvååker',
        idx: '35',
        name: 'Bjären, 432 76 Tvååker',
        lat: '57.056485',
        lon: '12.351331',
        type: 1,
      },
      {
        id: 'Björka, 432 77 Tvååker',
        idx: '36',
        name: 'Björka, 432 77 Tvååker',
        lat: '57.086932',
        lon: '12.592782',
        type: 1,
      },
      {
        id: 'Björkäng, 432 76 Tvååker',
        idx: '37',
        name: 'Björkäng, 432 76 Tvååker',
        lat: '57.008672',
        lon: '12.362631',
        type: 1,
      },
      {
        id: 'Björkängs camping, 432 76 Tvååker',
        idx: '38',
        name: 'Björkängs camping, 432 76 Tvååker',
        lat: '57.009606',
        lon: '12.350226',
        type: 1,
      },
      {
        id: 'Björkängs havsbad, 432 76 Tvååker',
        idx: '39',
        name: 'Björkängs havsbad, 432 76 Tvååker',
        lat: '57.009418',
        lon: '12.350792',
        type: 1,
      },
      {
        id: 'Björkängs kvarn, 432 76 Tvååker',
        idx: '40',
        name: 'Björkängs kvarn, 432 76 Tvååker',
        lat: '57.013822',
        lon: '12.357813',
        type: 1,
      },
      {
        id: 'Björkängsvägen, 432 78 Tvååker',
        idx: '41',
        name: 'Björkängsvägen, 432 78 Tvååker',
        lat: '57.039127',
        lon: '12.398336',
        type: 1,
      },
      {
        id: 'Björkedalsvägen, 432 77 Tvååker',
        idx: '42',
        name: 'Björkedalsvägen, 432 77 Tvååker',
        lat: '57.072153',
        lon: '12.369822',
        type: 1,
      },
      {
        id: 'Björktrastvägen, 432 76 Tvååker',
        idx: '43',
        name: 'Björktrastvägen, 432 76 Tvååker',
        lat: '57.010676',
        lon: '12.356177',
        type: 1,
      },
      {
        id: 'Bjurbäcken Braåsen, 688 91 Storfors',
        idx: '44',
        name: 'Bjurbäcken Braåsen, 688 91 Storfors',
        lat: '59.577204',
        lon: '14.211325',
        type: 1,
      },
      {
        id: 'Blidsbergs Påarp, 523 77 Blidsberg',
        idx: '45',
        name: 'Blidsbergs Påarp, 523 77 Blidsberg',
        lat: '57.941051',
        lon: '13.480934',
        type: 1,
      },
      {
        id: 'Blomvägen, 432 79 Tvååker',
        idx: '46',
        name: 'Blomvägen, 432 79 Tvååker',
        lat: '57.043586',
        lon: '12.401392',
        type: 1,
      },
      {
        id: 'Bocklerud Laåsen, 672 94 Årjäng',
        idx: '47',
        name: 'Bocklerud Laåsen, 672 94 Årjäng',
        lat: '59.393186',
        lon: '11.994550',
        type: 1,
      },
      {
        id: 'Bofinksvägen, 432 76 Tvååker',
        idx: '48',
        name: 'Bofinksvägen, 432 76 Tvååker',
        lat: '57.009660',
        lon: '12.353300',
        type: 1,
      },
      {
        id: 'Borrås, 432 76 Tvååker',
        idx: '49',
        name: 'Borrås, 432 76 Tvååker',
        lat: '57.014434',
        lon: '12.441340',
        type: 1,
      },
      {
        id: 'Bosgårdsvägen, 432 79 Tvååker',
        idx: '50',
        name: 'Bosgårdsvägen, 432 79 Tvååker',
        lat: '57.044970',
        lon: '12.399747',
        type: 1,
      },
    ],
    originLocations: [
      {
        id: '9021014014010000',
        idx: '1',
        name: 'Hålta kyrka, Kungälv',
        lat: '57.895206',
        lon: '11.827647',
        type: 0,
      },
      {
        id: '9021014012743000',
        idx: '2',
        name: 'Gårda, Mölndal',
        lat: '57.565347',
        lon: '12.092784',
        type: 0,
      },
      {
        id: '9021014002515000',
        idx: '3',
        name: 'Toftaåsgatan, Göteborg',
        lat: '57.645953',
        lon: '11.917170',
        type: 0,
      },
      {
        id: '9021014015934000',
        idx: '4',
        name: 'Hagaån, Orust',
        lat: '58.139155',
        lon: '11.625866',
        type: 0,
      },
      {
        id: '9021014021670000',
        idx: '5',
        name: 'Råane bro, Uddevalla',
        lat: '58.293814',
        lon: '11.984122',
        type: 0,
      },
      {
        id: '9021014021673000',
        idx: '6',
        name: 'Råane skola, Uddevalla',
        lat: '58.288996',
        lon: '11.988985',
        type: 0,
      },
      {
        id: '9021014021672000',
        idx: '7',
        name: 'Råane södra, Uddevalla',
        lat: '58.291360',
        lon: '11.986549',
        type: 0,
      },
      {
        id: 'A U Bååths väg, 439 94 Onsala',
        idx: '8',
        name: 'A U Bååths väg, 439 94 Onsala',
        lat: '57.390569',
        lon: '12.022587',
        type: 1,
      },
      {
        id: 'Allstakan Flaåsen, 671 98 Gunnarskog',
        idx: '9',
        name: 'Allstakan Flaåsen, 671 98 Gunnarskog',
        lat: '59.908834',
        lon: '12.510476',
        type: 1,
      },
      {
        id: 'Almvägen, 432 79 Tvååker',
        idx: '10',
        name: 'Almvägen, 432 79 Tvååker',
        lat: '57.044575',
        lon: '12.408116',
        type: 1,
      },
      {
        id: 'Alvägen, 432 79 Tvååker',
        idx: '11',
        name: 'Alvägen, 432 79 Tvååker',
        lat: '57.042903',
        lon: '12.404539',
        type: 1,
      },
      {
        id: 'Amandas väg, 432 77 Tvååker',
        idx: '12',
        name: 'Amandas väg, 432 77 Tvååker',
        lat: '57.060683',
        lon: '12.529336',
        type: 1,
      },
      {
        id: 'Anders Kristians väg, 432 77 Tvååker',
        idx: '13',
        name: 'Anders Kristians väg, 432 77 Tvååker',
        lat: '57.073115',
        lon: '12.368690',
        type: 1,
      },
      {
        id: 'Anemonvägen, 432 76 Tvååker',
        idx: '14',
        name: 'Anemonvägen, 432 76 Tvååker',
        lat: '57.012168',
        lon: '12.352572',
        type: 1,
      },
      {
        id: 'Angryd, 432 77 Tvååker',
        idx: '15',
        name: 'Angryd, 432 77 Tvååker',
        lat: '57.089359',
        lon: '12.540393',
        type: 1,
      },
      {
        id: 'Annelundsvägen, 432 76 Tvååker',
        idx: '16',
        name: 'Annelundsvägen, 432 76 Tvååker',
        lat: '57.042570',
        lon: '12.315941',
        type: 1,
      },
      {
        id: 'Askvägen, 432 79 Tvååker',
        idx: '17',
        name: 'Askvägen, 432 79 Tvååker',
        lat: '57.043370',
        lon: '12.403046',
        type: 1,
      },
      {
        id: 'Aspvägen, 432 79 Tvååker',
        idx: '18',
        name: 'Aspvägen, 432 79 Tvååker',
        lat: '57.044215',
        lon: '12.406894',
        type: 1,
      },
      {
        id: 'Augusts väg, 432 77 Tvååker',
        idx: '19',
        name: 'Augusts väg, 432 77 Tvååker',
        lat: '57.058598',
        lon: '12.526792',
        type: 1,
      },
      {
        id: 'Avavägen, 432 78 Tvååker',
        idx: '20',
        name: 'Avavägen, 432 78 Tvååker',
        lat: '57.047721',
        lon: '12.397797',
        type: 1,
      },
      {
        id: 'Rör, 432 77 Tvååker',
        idx: '21',
        name: 'Rör, 432 77 Tvååker',
        lat: '57.059937',
        lon: '12.445323',
        type: 1,
      },
      {
        id: 'Sik, 432 76 Tvååker',
        idx: '22',
        name: 'Sik, 432 76 Tvååker',
        lat: '57.001220',
        lon: '12.367845',
        type: 1,
      },
      {
        id: 'Påarps gård, Håcksvik',
        idx: '23',
        name: 'Påarps gård, Håcksvik',
        lat: '57.315473',
        lon: '13.147488',
        type: 1,
      },
      {
        id: 'Ängarna, 432 76 Tvååker',
        idx: '24',
        name: 'Ängarna, 432 76 Tvååker',
        lat: '57.028376',
        lon: '12.375692',
        type: 1,
      },
      {
        id: 'Ängvägen, 432 78 Tvååker',
        idx: '25',
        name: 'Ängvägen, 432 78 Tvååker',
        lat: '57.047217',
        lon: '12.402363',
        type: 1,
      },
      {
        id: 'Ästad, 432 77 Tvååker',
        idx: '26',
        name: 'Ästad, 432 77 Tvååker',
        lat: '57.066535',
        lon: '12.562155',
        type: 1,
      },
      {
        id: 'Ästad bondgård, 432 77 Tvååker',
        idx: '27',
        name: 'Ästad bondgård, 432 77 Tvååker',
        lat: '57.063245',
        lon: '12.555953',
        type: 1,
      },
      {
        id: 'Bäckvägen, 432 79 Tvååker',
        idx: '28',
        name: 'Bäckvägen, 432 79 Tvååker',
        lat: '57.038741',
        lon: '12.404889',
        type: 1,
      },
      {
        id: 'Ballstorp Kullaås, 535 93 Kvänum',
        idx: '29',
        name: 'Ballstorp Kullaås, 535 93 Kvänum',
        lat: '58.191256',
        lon: '13.253812',
        type: 1,
      },
      {
        id: 'Barrvägen, 432 79 Tvååker',
        idx: '30',
        name: 'Barrvägen, 432 79 Tvååker',
        lat: '57.035675',
        lon: '12.405806',
        type: 1,
      },
      {
        id: 'Beckasinvägen, 432 76 Tvååker',
        idx: '31',
        name: 'Beckasinvägen, 432 76 Tvååker',
        lat: '57.007296',
        lon: '12.358487',
        type: 1,
      },
      {
        id: 'Bergsjö Lövbodaåsarna, 681 94 Kristinehamn',
        idx: '32',
        name: 'Bergsjö Lövbodaåsarna, 681 94 Kristinehamn',
        lat: '59.421358',
        lon: '14.248622',
        type: 1,
      },
      {
        id: 'Bjällemad, 432 79 Tvååker',
        idx: '33',
        name: 'Bjällemad, 432 79 Tvååker',
        lat: '57.037329',
        lon: '12.411271',
        type: 1,
      },
      {
        id: 'Bjällemadvägen, 432 79 Tvååker',
        idx: '34',
        name: 'Bjällemadvägen, 432 79 Tvååker',
        lat: '57.037662',
        lon: '12.406678',
        type: 1,
      },
      {
        id: 'Bjären, 432 76 Tvååker',
        idx: '35',
        name: 'Bjären, 432 76 Tvååker',
        lat: '57.056485',
        lon: '12.351331',
        type: 1,
      },
      {
        id: 'Björka, 432 77 Tvååker',
        idx: '36',
        name: 'Björka, 432 77 Tvååker',
        lat: '57.086932',
        lon: '12.592782',
        type: 1,
      },
      {
        id: 'Björkäng, 432 76 Tvååker',
        idx: '37',
        name: 'Björkäng, 432 76 Tvååker',
        lat: '57.008672',
        lon: '12.362631',
        type: 1,
      },
      {
        id: 'Björkängs camping, 432 76 Tvååker',
        idx: '38',
        name: 'Björkängs camping, 432 76 Tvååker',
        lat: '57.009606',
        lon: '12.350226',
        type: 1,
      },
      {
        id: 'Björkängs havsbad, 432 76 Tvååker',
        idx: '39',
        name: 'Björkängs havsbad, 432 76 Tvååker',
        lat: '57.009418',
        lon: '12.350792',
        type: 1,
      },
      {
        id: 'Björkängs kvarn, 432 76 Tvååker',
        idx: '40',
        name: 'Björkängs kvarn, 432 76 Tvååker',
        lat: '57.013822',
        lon: '12.357813',
        type: 1,
      },
      {
        id: 'Björkängsvägen, 432 78 Tvååker',
        idx: '41',
        name: 'Björkängsvägen, 432 78 Tvååker',
        lat: '57.039127',
        lon: '12.398336',
        type: 1,
      },
      {
        id: 'Björkedalsvägen, 432 77 Tvååker',
        idx: '42',
        name: 'Björkedalsvägen, 432 77 Tvååker',
        lat: '57.072153',
        lon: '12.369822',
        type: 1,
      },
      {
        id: 'Björktrastvägen, 432 76 Tvååker',
        idx: '43',
        name: 'Björktrastvägen, 432 76 Tvååker',
        lat: '57.010676',
        lon: '12.356177',
        type: 1,
      },
      {
        id: 'Bjurbäcken Braåsen, 688 91 Storfors',
        idx: '44',
        name: 'Bjurbäcken Braåsen, 688 91 Storfors',
        lat: '59.577204',
        lon: '14.211325',
        type: 1,
      },
      {
        id: 'Blidsbergs Påarp, 523 77 Blidsberg',
        idx: '45',
        name: 'Blidsbergs Påarp, 523 77 Blidsberg',
        lat: '57.941051',
        lon: '13.480934',
        type: 1,
      },
      {
        id: 'Blomvägen, 432 79 Tvååker',
        idx: '46',
        name: 'Blomvägen, 432 79 Tvååker',
        lat: '57.043586',
        lon: '12.401392',
        type: 1,
      },
      {
        id: 'Bocklerud Laåsen, 672 94 Årjäng',
        idx: '47',
        name: 'Bocklerud Laåsen, 672 94 Årjäng',
        lat: '59.393186',
        lon: '11.994550',
        type: 1,
      },
      {
        id: 'Bofinksvägen, 432 76 Tvååker',
        idx: '48',
        name: 'Bofinksvägen, 432 76 Tvååker',
        lat: '57.009660',
        lon: '12.353300',
        type: 1,
      },
      {
        id: 'Borrås, 432 76 Tvååker',
        idx: '49',
        name: 'Borrås, 432 76 Tvååker',
        lat: '57.014434',
        lon: '12.441340',
        type: 1,
      },
      {
        id: 'Bosgårdsvägen, 432 79 Tvååker',
        idx: '50',
        name: 'Bosgårdsvägen, 432 79 Tvååker',
        lat: '57.044970',
        lon: '12.399747',
        type: 1,
      },
    ],
  },
}
