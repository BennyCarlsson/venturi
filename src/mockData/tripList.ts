import { LocationType } from 'redux/locationsSlice'
import { TripListState } from 'redux/tripListSlice'

export const mockTripList: TripListState = {
  trips: [
    {
      name: 'TestName',
      destination: {
        id: '1',
        idx: '1',
        lat: '1',
        lon: '1',
        name: 'Destination Name',
        type: LocationType.CoordLocation,
      },
      origin: {
        id: '2',
        idx: '1',
        lat: '2',
        lon: '2',
        name: 'Origin name',
        type: LocationType.CoordLocation,
      },
    },
    {
      name: 'TestName2',
      destination: {
        id: '1',
        idx: '1',
        lat: '1',
        lon: '1',
        name: 'Destination Name',
        type: LocationType.CoordLocation,
      },
      origin: {
        id: '2',
        idx: '1',
        lat: '2',
        lon: '2',
        name: 'Origin name',
        type: LocationType.CoordLocation,
      },
    },
    {
      name: 'TestName3',
      destination: {
        id: '1',
        idx: '1',
        lat: '1',
        lon: '1',
        name: 'Destination Name',
        type: LocationType.CoordLocation,
      },
      origin: {
        id: '2',
        idx: '1',
        lat: '2',
        lon: '2',
        name: 'Origin name',
        type: LocationType.CoordLocation,
      },
    },
  ],
}
