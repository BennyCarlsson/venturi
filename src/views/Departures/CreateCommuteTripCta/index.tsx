import Colors from 'tokens/colors'
import {
  CreateCommuteTripCtaWrapper,
  StyledParagraph,
  StyledPlusIcon,
} from './styles'

type CreateCommuteTripCtaProps = {
  onClick: () => void
}

const CreateCommuteTripCta = ({ onClick }: CreateCommuteTripCtaProps) => {
  return (
    <CreateCommuteTripCtaWrapper
      onClick={(e) => {
        e.preventDefault()
        onClick()
      }}
    >
      <StyledPlusIcon />
      <StyledParagraph fontSize={18} color={Colors.white}>
        Create commute trip
      </StyledParagraph>
    </CreateCommuteTripCtaWrapper>
  )
}

export default CreateCommuteTripCta
