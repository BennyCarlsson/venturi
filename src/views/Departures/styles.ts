import styled from 'styled-components'

export const DeparturesWrapper = styled.div`
  padding: 24px;
`
