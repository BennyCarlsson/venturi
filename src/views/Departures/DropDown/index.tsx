import { Fragment } from 'react';
import { Location } from 'redux/locationsSlice';
import { Li, Ul } from './styles';

type DropDownProps = {
  locations: Location[] | undefined;
  setInput: (value: React.SetStateAction<string>) => void;
  setLocation: (value: React.SetStateAction<Location | null>) => void;
  show: boolean;
};

const DropDown = ({ locations, setInput, setLocation, show }: DropDownProps) => {
  return (
    <Fragment>
      {show && (
        <Ul>
          {locations?.map((loc) => (
            <Li
              key={`key-${loc.id}`}
              onClick={(e) => {
                e.preventDefault();
                console.log(loc.id);
                setInput(loc.name);
                setLocation(loc);
              }}
            >
              {loc.name}
            </Li>
          ))}
        </Ul>
      )}
    </Fragment>
  );
};

export default DropDown;
