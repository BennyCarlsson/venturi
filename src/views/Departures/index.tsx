import { showDialog } from 'redux/dialogSlice'
import CreateCommuteTripCta from './CreateCommuteTripCta'
import Heading from 'components/Heading'
import { DeparturesWrapper } from './styles'
import { useAppDispatch, useAppSelector } from 'hooks/redux'
import { useEffect } from 'react'
import { saveTripList } from 'utils/localStorage'

const Departures = () => {
  const dispatch = useAppDispatch()
  const tripList = useAppSelector((state) => state.tripList)

  useEffect(() => {
    saveTripList(tripList)
  }, [tripList])

  return (
    <DeparturesWrapper>
      <Heading headingType='h2'>Departures</Heading>
      {tripList.trips.map((trip, i) => (
        <div key={i}>
          --
          <p>{trip.destination?.name}</p>
          <p>{trip.origin?.name}</p>
          ---
        </div>
      ))}
      <CreateCommuteTripCta
        onClick={() => {
          dispatch(showDialog())
        }}
      />
    </DeparturesWrapper>
  )
}

export default Departures
