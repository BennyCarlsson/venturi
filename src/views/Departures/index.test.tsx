import renderer from 'react-test-renderer'
import Departures from '.'
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { useSelector, useDispatch } from 'react-redux'
import { mockTripList } from 'mockData/tripList'

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}))

describe('<Departures />', () => {
  it('should match snapshot', () => {
    ;(useSelector as jest.Mock).mockReturnValueOnce(mockTripList)
    ;(useDispatch as jest.Mock).mockReturnValue(() => {})
    const tree = renderer.create(<Departures />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
