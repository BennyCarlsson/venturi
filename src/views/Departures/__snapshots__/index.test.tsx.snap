// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Departures /> should match snapshot 1`] = `
<div
  className="sc-jrAGrp hffLVH"
>
  <h2
    className="sc-jSgupP ffTEWZ"
  >
    Departures
  </h2>
  <div>
    --
    <p>
      Destination Name
    </p>
    <p>
      Origin name
    </p>
    ---
  </div>
  <a
    className="sc-gsTCUz jxAhtz"
    onClick={[Function]}
  >
    <svg
      className="sc-dlfnbm ilDCCC"
    >
      plus.svg
    </svg>
    <p
      className="sc-bdfBwQ gbRIVK sc-hKgILt SVDlr"
      color="#FFFFFF"
      fontSize={18}
    >
      Create commute trip
    </p>
  </a>
</div>
`;
